kernel/kernel-api/src/main/java/io/delta/kernel/ResolvedTableBuilder.java

```diff
@@ -18,18 +18,34 @@
 
 import io.delta.kernel.annotation.Experimental;
 import io.delta.kernel.engine.Engine;
+import io.delta.kernel.internal.actions.Metadata;
+import io.delta.kernel.internal.actions.Protocol;
 import io.delta.kernel.internal.files.ParsedLogData;
+import io.delta.kernel.internal.files.ParsedLogData.ParsedLogType;
 import java.util.List;
 
+/**
+ * Builder for constructing a {@link ResolvedTable} instance.
+ *
+ * <p>This builder allows table managers (filesystems, catalogs) to provide any information they may
+ * know about a Delta table and get back a {@link ResolvedTable}. When {@link #build(Engine)} is
+ * invoked, Kernel will automatically fill any missing information needed to construct the {@link
+ * ResolvedTable} by reading from the filesystem as needed.
+ *
+ * <p>If no version is specified, the builder will resolve to the latest version. Depending on the
+ * {@link ParsedLogData} provided, Kernel can avoid expensive filesystem operations.
+ */
 @Experimental
 public interface ResolvedTableBuilder {
   ResolvedTableBuilder atVersion(long version);
 
   // TODO: atTimestamp
 
+  /** For now, only log datas of type {@link ParsedLogType#RATIFIED_STAGED_COMMIT}s are supported */
   ResolvedTableBuilder withLogData(List<ParsedLogData> logData);
 
-  // TODO: withProtocolAndMetadata
+  // TODO: P & M must be public interfaces, not internal classes
+  ResolvedTableBuilder withProtocolAndMetadata(Protocol protocol, Metadata metadata);
```

Comment:
Good with this, but there is an alternative: not couple P&M (I know they are mentioned almost altogether but conceptually they are two concept)
e.g.
ResolvedTableBuilder withProtocol
ResolvedTableBuilder withMetadata